pkg/daemon/server/handle.go

```diff
@@ -100,20 +113,17 @@ func (cdh *cniDaemonHandler) handleAdd(req *restful.Request, resp *restful.Respo
 		time.Sleep(backOffBase)
 		backOffBase = backOffBase * 2
 
-		pod := &corev1.Pod{}
-		if err := cdh.mgrAPIReader.Get(context.TODO(), types.NamespacedName{
-			Name:      podRequest.PodName,
-			Namespace: podRequest.PodNamespace,
-		}, pod); err != nil {
-			errMsg := fmt.Errorf("failed to get pod %v/%v: %v", podRequest.PodName, podRequest.PodNamespace, err)
+		ipFamily := ipamtypes.ParseIPFamilyFromString(pod.Annotations[constants.AnnotationIPFamily])
```

Comment:
can you parse ip family outside the loop?

Reply:
fixed